File: Python3/Medium/RuntimeError/maximum-ice-cream-bars_1.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        
        # Counting sort
        sorted_costs = [0] * (max(costs) + 1)
        for cost in costs:
            sorted_costs[cost] += 1
        count = 0
        for i in range(len(sorted_costs)):
            if coins >= i:
                if sorted_costs[i] <= coins // i:
                    count += sorted_costs[i]
                    coins -= sorted_costs[i] * i
                else:
                    count += coins // i
                    break
        return count

File: Rust/Medium/CompileError/maximum-ice-cream-bars_1.rs
impl Solution {
    pub fn max_ice_cream(costs: Vec<i32>, coins: i32) -> i32 {
        
        let mut costs = costs;
        costs.sort();
        let mut res = 0;
        for cost in costs {
            if coins < cost {
                break;
            }
            res += 1;
            coins -= cost;
        }
        res
    }
}